sklearn/tree/tests/test_tree.py

```diff
@@ -2785,21 +2785,27 @@ def test_regression_tree_missing_values_toy(Tree, X, criterion, global_random_se
     https://github.com/scikit-learn/scikit-learn/issues/28316
     """
     X = X.reshape(-1, 1)
-    y = np.arange(6)
+    y = np.arange(1, 7)
+    # y needs to be > 0 for this test to pass with poisson criterion
+    # Poisson criterion allows multiple optimal splits when y[i] = 0 because
+    # the impurity calculation involves log(y[i]), which is undefined at 0.
```

Comment:
This is not entirely correct. And for me, it does not relate to the next sentence.